feat(v2): Validation failures return 400s (#1489)
aws-powertools/powertools-lambda-java: docs/utilities/validation.md

`@Validation` annotation is used to validate either inbound events or functions' response.

> It will fail fast if an event or response doesn't conform with given JSON Schema. For most type of events a `ValidationException` will be thrown. But for 

Reviewer: ```suggestion
It will fail fast if an event or response doesn't conform with given JSON Schema. For most type of events a `ValidationException` will be thrown. 
```

> specific `APIGatewayProxyRequestEvent` and `APIGatewayV2HTTPEvent`, instead, the `@Validation` annotation will build and return a custom

Reviewer: ```suggestion
For API gateway events associated with REST APIs and HTTP APIs -  `APIGatewayProxyRequestEvent` and `APIGatewayV2HTTPEvent` - the `@Validation` annotation will build and return a custom 400 / "Bad Request" response, with a body containing the validation errors. This saves you from having to catch the validation exception and map it back to a meaningful user error yourself.
```

Reviewer: Trying to make it explicit what is happening and why!

> 400 (BAD_REQUEST) response. Its body will contain the validation error(s) message(s).

Reviewer: ```suggestion
```
